client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/constraint/H2ConstraintExtractor.java

```diff
@@ -0,0 +1,132 @@
+        allConstraints.addAll(columnConstraints);
+        allConstraints.addAll(tableCheckExpressions);
+        return allConstraints;
+    }
+
+    /**
+     * For each table in the schema DTO, this method appends
+     * the constraints that are originated in the ALTER TABLE commands
+     * for those particular tables.
+     * <p>
+     * Foreign keys are handled separately in the JDBC metadata
+     *
+     * @param connectionToH2 a connection to a H2 database
+     * @param schemaDto
+     * @throws SQLException if the connection to the H2 database fails,
+     */
+    private List<DbTableConstraint> extractTableConstraints(Connection connectionToH2, DbSchemaDto schemaDto) throws SQLException {
+
+        List<DbTableConstraint> tableCheckExpressions = new LinkedList<>();
+
+        String tableSchema = schemaDto.name;
+        for (TableDto tableDto : schemaDto.tables) {
+            String tableName = tableDto.name;
+            Statement statement = connectionToH2.createStatement();
+            ResultSet constraints = statement.executeQuery("Select * From INFORMATION_SCHEMA.CONSTRAINTS "
+                    + " where CONSTRAINTS.TABLE_SCHEMA='" + tableSchema + "' and CONSTRAINTS.TABLE_NAME='" + tableName + "'");
+
+            while (constraints.next()) {
+                String constraintType = constraints.getString("CONSTRAINT_TYPE");
+                String sqlCheckExpression = constraints.getString("CHECK_EXPRESSION");
+                String columnList = constraints.getString("COLUMN_LIST");
+
+                if (constraintType.equals("UNIQUE")) {
+                    assert (sqlCheckExpression == null);
+                    List<String> uniqueColumnNames = Arrays.stream(columnList.split(",")).map(String::trim).collect(Collectors.toList());
+                    DbTableUniqueConstraint uniqueConstraint = new DbTableUniqueConstraint(tableName, uniqueColumnNames);
+                    tableCheckExpressions.add(uniqueConstraint);
+
+                } else if (constraintType.equals("REFERENTIAL")) {
+                    /**
+                     * This type of constraint is already handled by
+                     * JDBC Metadata
+                     **/
+                    continue;
+                } else if (constraintType.equals("PRIMARY KEY") || constraintType.equals("PRIMARY_KEY")) {
+                    /**
+                     * This type of constraint is already handled by
+                     * JDBC Metadata
+                     **/
+                    continue;
+                } else if (constraintType.equals("CHECK")) {
+                    assert (columnList == null);
+
+                    DbTableCheckExpression constraint = new DbTableCheckExpression(tableName, sqlCheckExpression);
+                    tableCheckExpressions.add(constraint);
+
+                } else {
+                    throw new RuntimeException("Unknown constraint type : " + constraintType);
+                }
```

Comment:
not sure we should throw an exception here, which might stop EM from working. We could still be able to generate test cases even if cannot handle all possible constraints in different SUTs.
See for example how `HeuristicsCalculator.cannotHandle` works

Reply:
Understood. I will add a "cannotHandle" warning when extracting unsupported constraints in both H2 and Postgres